### ISSUE TITLE
Install failing on Olimex Lime 2

### ISSUE THREAD

**Issue by nicdall** [YunoHost-Apps/kresus_ynh#161]
### Describe the bug

*Install is failing on Olimex Lime2. From the logs it seems to be due to the cryptography package, but I am not sure how to proceed with the debugging.*

*I saw that Kresus does not support 32 bit, but the installation does not seem to be blocked by that. Any help understanding this issue is appreciated.*

### Context

- Hardware: *Olimex Lime 2*
- YunoHost version: 11.1.22
- I have access to my server: *Through SSH | through the webadmin*
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: *no*

### Steps to reproduce

```sh
sudo yunohost app install kresus
```

### Logs

https://paste.yunohost.org/raw/xaxicuvulo

**Comment by nicofrand** [comment]
Hi,

could you try installing `libssl-dev` with apt and then try again please?

**Comment by nicdall** [comment]
Hi,

I installed `libssl-dev` successfully.
However the install is still failing and the log is now empty : https://paste.yunohost.org/raw/wobalonimi

**Comment by nicofrand** [comment]
Uh, without logs I cannot know what still fails…

**Comment by nicdall** [comment]
Of course. I tried again an this time the logs is saved correctly : https://paste.yunohost.org/raw/ogiqanafuh

The error seems to be `No space left on device (os error 28)`

The result of `df -h` is :
```
Filesystem      Size  Used Avail Use% Mounted on
udev            447M     0  447M   0% /dev
tmpfs           100M   11M   89M  11% /run
/dev/mmcblk0p1   29G  7.8G   21G  28% /
tmpfs           500M   76K  500M   1% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
/dev/sda1       1.8T   95G  1.7T   6% /mnt/hdd
tmpfs           500M  1.1M  499M   1% /tmp
/dev/zram0       49M   13M   33M  29% /var/log
tmpfs           100M     0  100M   0% /run/user/0
tmpfs           100M     0  100M   0% /run/user/22311
```

Do I need to increase the space for `/tmp`? Or is there something else I did not see?

**Comment by nicofrand** [comment]
Ah bin là t'as juste plus de place sur ton disque dur/SSD/partition !

Le 26 juillet 2023 21:22:00 GMT+02:00, nicdall ***@***.***> a écrit :

**Comment by nicofrand** [comment]
Try lsblk ?

Le 26 juillet 2023 21:22:00 GMT+02:00, nicdall ***@***.***> a écrit :

**Comment by nicofrand** [comment]
You might want to check https://www.maketecheasier.com/fix-linux-no-space-left-on-device-error/ too.
Maybe also restart the device to clean up deleted files

**Comment by nicofrand** [comment]
What's the result of `sudo df -h /var/log`?
I hear some distros put it in ram...

**Comment by nicdall** [comment]
Result of `lsblk`:
```
NAME         MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
sda            8:0    0   1.8T  0 disk
└─sda1         8:1    0   1.8T  0 part /mnt/hdd
mmcblk0      179:0    0  29.1G  0 disk
└─mmcblk0p1  179:1    0  28.8G  0 part /
mmcblk1      179:8    0   3.6G  0 disk
└─mmcblk1p1  179:9    0   3.6G  0 part
mmcblk1boot0 179:16   0    16M  1 disk
mmcblk1boot1 179:24   0    16M  1 disk
zram0        254:0    0    50M  0 disk /var/log
zram1        254:1    0 499.5M  0 disk [SWAP]
```

Result of sudo df -h /var/log:
```
Filesystem      Size  Used Avail Use% Mounted on
/dev/zram0       49M   12M   33M  27% /var/log
```

I tried following the tutorial but could see any change.
I tried logging the space in /tmp and it seems to be the one getting full during install (unfortunately I do not have the logs anymore)

**Comment by nicofrand** [comment]
OK, I don't know how to fix your RAM issue then. Maybe with some bigger swap?

**Comment by nicdall** [comment]
Thank you for the suggestions. Searching around I found the suggestion of remounting the /tmp partition with bigger space: `mount -o remount,size=2G /tmp`.
It seemed to have worked, as the install is going farther, but it is know failing later: https://paste.yunohost.org/raw/abihebikip 
Do I need to increase the size of the tmp partition more? Or is this unrelated?

**Comment by nicdall** [comment]
I have not been disconnected, I do not have any firewall in place aside from the one coming with Yunohost, and I haven't had any connection issue in the past. 
I will retry installation now.

**Comment by nicdall** [comment]
New log, same problem: https://paste.yunohost.org/raw/xarilayonu

Based on [this](https://forum.yunohost.org/t/impossible-dinstaller-peertube/6557/7) answer on the Yunohost website, i tried `curl -JOLs https://registry.yarnpkg.com/date-fns/-/date-fns-2.29.3.tgz` and it is working correctly (the file is downloaded as expected).

**Comment by bnjbvr** [comment]
What does `/var/www/kresus/yarn-error.log` contain?

**Comment by nicdall** [comment]
Unfortunately it is deleted after the failed installation. I can try to deactivate the removal of the app, but I expect it to be the same as the error that is outputed "https://registry.yarnpkg.com/date-fns/-/date-fns-2.29.3.tgz: ESOCKETTIMEDOUT"

**Comment by nicdall** [comment]
I tried re running the installation with --no-remove-on-failure to get the content of the [yarn-error.log file](https://paste.yunohost.org/raw/ukofeginab).

I also tried installing yarn separately and increasing the timeout with `sudo yarn config set network-timeout 500000` but the error remains the same.

**Comment by nicofrand** [comment]
Can you try:

```bash
mkdir /tmp/testyarndatefns
cd /tmp/testyarndatefns
yarn init -y
yarn add date-fns@2.29.3  --verbose 
```

And tell us the output?

Then 

```bash
yarn remove date-fns --verbose 
yarn add date-fns@2.29.3 --network-timeout 60000 --verbose 
```

**Comment by nicdall** [comment]
From the first set of command, the output is huge (did not get the full one) but the last lines are :
```
verbose 106.591926408 Copying "/home/nicdal/.cache/yarn/v6/npm-date-fns-2.29.3-27402d2fc67eb442b511b70bbdf98e6411cd68a8-integrity/node_modules/date-fns/esm/locale/zh-TW/_lib/match/index.js" to "/tmp/testyarndatefns/node_modules/date-fns/esm/locale/zh-TW/_lib/match/index.js".
[4/4] Building fresh packages...
success Saved lockfile.
success Saved 1 new dependency.
info Direct dependencies
└─ date-fns@2.29.3
info All dependencies
└─ date-fns@2.29.3
Done in 103.88s.
```

And here is the output of the second section : https://paste.yunohost.org/raw/abocuyabiv

**Comment by nicofrand** [comment]
Ok so this times it works. I wonder if that could still be a RAM issue when downloading the file after already storing a bunch of things in RAM...

**Comment by nicdall** [comment]
My SWAP file is on an external SSD, perhaps this is causing the issue. I found this [issue ](https://github.com/yarnpkg/yarn/issues/8242) on yarn's Github which seems to be related, however I do not really understand how to fix it.

**Comment by nicofrand** [comment]
Closing as this seems to come from yarn
